storage/google/cloud/storage/blob.py

```diff
@@ -751,16 +943,17 @@ def upload_from_filename(self, filename, content_type=None, client=None):
         :type content_type: str
         :param content_type: Optional type of content being uploaded.
 
-        :type client: :class:`~google.cloud.storage.client.Client` or
-                      ``NoneType``
-        :param client: Optional. The client to use.  If not passed, falls back
+        :type client: :class:`~google.cloud.storage.client.Client`
+        :param client: (Optional) The client to use.  If not passed, falls back
                        to the ``client`` stored on the blob's bucket.
         """
         content_type = self._get_content_type(content_type, filename=filename)
 
         with open(filename, 'rb') as file_obj:
+            total_bytes = os.fstat(file_obj.fileno()).st_size
```

Comment:
Not sure. I can factor this out as a `_get_size_from_fileno` helper and then just return `None` on `OSError` / `UnsupportedOperation`? (Seems YAGNI, but I would do it.)